lib/src/disposable/disposable.dart

```diff
@@ -214,19 +219,17 @@ class Disposable implements _Disposable, DisposableManagerV3 {
     }
     _isDisposing = true;
 
-    List<Future<dynamic>> futures = []
-      ..addAll(_internalDisposables.map((disposable) => disposable.dispose()))
-      ..add(onDispose());
-    _internalDisposables = [];
+    await Future.wait(_awaitableFutures);
+    _awaitableFutures.clear();
+
+    for (var disposable in _internalDisposables) {
+      await disposable.dispose();
+    }
```

Comment:
:man_shrugging: I'd be interested in seeing the change in the JS output to see what is changing under the hood.